### DISCUSSION TITLE
NVIDIA® Jetson AGX Orin 32GB H01 Kit | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#474] [Q&A] [answered]
# NVIDIA® Jetson AGX Orin 32GB H01 Kit | Seeed Studio Wiki

Flash JetPack to NVIDIA® Jetson AGX Orin 32GB H01 Kit

https://wiki.seeedstudio.com/Jetson_AGX_Orin_32GB_H01_Flash_Jetpack/

**Comment by jinchenglee** [comment]
May I apply the board driver to newer versions of Jetpack? Say v5.1.1? Is there any documentations/explanations on the board driver files? Thanks.

**Reply by MatthewJeffson** [reply]
My apology for the late response. The newer version of Jetpack isn't ready yet. :D

**Reply by MatthewJeffson** [reply, marked as answer]
Hello! The Jetpack 5.1.1 is ready now and the wiki will be updating soon as well.
[605_jp5.1.1_R35.3.1.zip](https://github.com/Seeed-Studio/wiki-documents/files/12271738/605_jp5.1.1_R35.3.1.zip)

**Reply by jinchenglee** [reply]
Can you summarize what was the required changes so that hobbyist can try do update themselves when nvidia rolls out new version?

**Reply by MatthewJeffson** [reply]
> Can you summarize what was the required changes so that hobbyist can try do update themselves when nvidia rolls out new version?

Hi, I have asked some of my colleagues about it these two days, and unfortunately we can't provide the patch yet. All we can present are on the [product page](https://www.seeedstudio.com/AGX-Orin-32GB-H01-Kit-p-5569.html) and the [wiki page](https://wiki.seeedstudio.com/Jetson_AGX_Orin_32GB_H01_Flash_Jetpack/).

Feel free to check our new page about [products based on NVIDIA Jetson modules](https://wiki.seeedstudio.com/NVIDIA_Jetson/).

And I will be honored if you can share with us some of your projects or ideas using NVIDIA Jetson devices.

Have a nice day!

**Comment by shichenliu97** [comment]
Hi, sorry if there is mentioned somewhere. But how do you remove the bottom plate of the Orin H01 to install an m.2 drive?

**Reply by jinchenglee** [reply]
I don't open my Orin H01 kit. I believe removing the rubber footings and you'll see the holes through which you can access the screws. A philips screw drive can help you split the top from the bottom.

**Reply by KoreTeknology** [reply]
> I don't open my Orin H01 kit. I believe removing the rubber footings and you'll see the holes through which you can access the screws. A philips screw drive can help you split the top from the bottom.

This is not a philips screw (under the foot) but an Allen Key standard! i was also surprised, i spent 2 days looking for this info with the fer to break the box, as i wanted to open it to install my SSD drive inside!

**Comment by jjannik** [comment]
Where is the  Debug UART Port for this device ?

**Reply by xzxcessarr** [reply]
Hello, if you wanna use debug mode for your project, you can use the output of terminal which ubuntu including or using the type-c interface near the GPIO interface to connect with your computer, set the brud rate at 9600 or 115200

**Reply by tomtana** [reply]
do you mean by this that ttyUSB* will be mounted when the USB cable is attached? If so, in my case it is not visible. Any suggestions?

**Reply by imed-twosix** [reply]
did you ever figure this out?

**Comment by FabianJohnThl** [comment]
Solved an issue with this Tutorial:
I got the Error: invalid target board on performing flash.sh

When I copied the command from this Tutorial: `sudo ./flash.sh jetson‐agx‐orin‐devkit mmcblk0p1` this error occurred. It seems that the - in jetson‐agx‐orin‐devkit are the problem. When I replace them in the Terminal after copying, everything works fine. 

I hope this helps others not to waste time.

**Comment by ghost** [comment]
Nvidia released developer preview of Jetpack 6.0 with L4T version 36.2.0. Please let me know when you are planning on supporting the new version.

**Reply by daniel-amag** [reply]
@MatthewJeffson Any word on this 6.0 support? It's slated for production release March 2024 so it should be out of developer preview pretty soon.

**Reply by ghost** [reply]
Jetpack 6.0 is released. https://developer.nvidia.com/embedded/jetson-linux-r363 

Please let us know the timeline for update to L4T 36.3.0

**Comment by ghost** [comment]
Also you are distributing kernel/Image without reference to its source. Could you also please publish the source code used to generate the kernel image.

**Comment by nicholas-presien** [comment]
@MatthewJeffson does the peripheral drivers for L4T35.3.1 works with L4T35.5.0?

**Comment by ghost** [comment]
Connecting a usb wifi dongle and trying to connect to wifi hangs the system immediately. Reboot is the only option. I am using Netgear A6210 USB wifi adapter.

**Reply by ghost** [reply]
[ 7790.738849] usb 2-3.4: new SuperSpeed Gen 1 USB device number 3 using tegra-xusb
[ 7790.762866] usb 2-3.4: New USB device found, idVendor=0846, idProduct=9053, bcdDevice= 1.00
[ 7790.762874] usb 2-3.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 7790.762878] usb 2-3.4: Product: A6210
[ 7790.762881] usb 2-3.4: Manufacturer: NETGEAR
[ 7790.762884] usb 2-3.4: SerialNumber: 100
[ 7790.931217] usb 2-3.4: reset SuperSpeed Gen 1 USB device number 3 using tegra-xusb
[ 7790.954798] mt76x2u 2-3.4:1.0: ASIC revision: 76120044
[ 7791.055371] mt76x2u 2-3.4:1.0: ROM patch build: 20141115060606a
[ 7791.253818] mt76x2u 2-3.4:1.0: Firmware Version: 0.0.00
[ 7791.253825] mt76x2u 2-3.4:1.0: Build: 1
[ 7791.253828] mt76x2u 2-3.4:1.0: Build Time: 201507311614____
[ 7792.207831] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[ 7792.211905] usbcore: registered new interface driver mt76x2u
[ 7853.203349] wlan1: authenticate with fe:9e:38:da:f3:c7
[ 7853.823460] wlan1: send auth to fe:9e:38:da:f3:c7 (try 1/3)
[ 7853.825549] wlan1: authenticated
[ 7853.832511] wlan1: associate with fe:9e:38:da:f3:c7 (try 1/3)
[ 7853.842050] wlan1: RX AssocResp from fe:9e:38:da:f3:c7 (capab=0x1511 status=0 aid=8)
[ 7853.857242] wlan1: associated
[ 7853.870459] wlan1: Limiting TX power to 30 (30 - 0) dBm as advertised by fe:9e:38:da:f3:c7
[ 7853.900039] IPv6: ADDRCONF(NETDEV_CHANGE): wlan1: link becomes ready

**Reply by ghost** [reply]
I have also confirmed that TP-Link USB adapter with RTL8814AU driver works without issues.

**Comment by GJSea** [comment]
I followed all of the instructions above and successfully flashed everything, reaching the final  "Coldbooting the device, *** The target t186ref has been flashed successfully. *** Reset the board to boot from internal eMMC."

However when I do reboot the board/powerup, the device seems to automatically go into Recovery mode again and seems to be bricked. The typical NVidia splash/BIOS screen is never displayed. No /dev/ttyUSB are listed on the host either, so I'm not able to get console output to diagnose any booting errors.

Any pointers? Thanks!

**Reply by daniel-amag** [reply]
Whenever I had symptoms like that it was because I had not applied the peripheral drivers correctly. Here are the steps I eventually puzzled out to get the flashing working repeatedly for me:

1. Install Nvidia SDK Manager on host machine (may need to be Ubuntu 20 or 18)
2. Get (“Download without install” toggle) JetPack 5.1.1   DO NOT GET ANY OTHER VERSION.
3. Download [Seeed Studio drivers](https://sourceforge.net/projects/nvidia-jetson/files/Jetson-AGX-Orin-32GB-H01-Kit/Driver-for-JP-5.1.1/AGX-Orin-32GB-H01-JP5.1.1.zip/download)
4. Go into the Nvidia SDK Manager install directory, usually:
 `cd ~/nvidia/nvidia_sdk/JetPack_5.1.1_Linux_JETSON_AGX_ORIN_TARGETS`
5. `sudo ./apply_binaries.sh`
6. Copy Seeed Studio driver zip file here and unzip, replacing all
7. Put device into force recovery mode by:
    a. Powering it off by holding the power button ![image](https://github.com/Seeed-Studio/wiki-documents/assets/92066886/4f28f657-b8e9-4209-816a-2a967e269a35) until there’s just a single red LED
    b. Press and hold the middle button :arrows_counterclockwise: and power on by pressing the power button ![image](https://github.com/Seeed-Studio/wiki-documents/assets/92066886/4f28f657-b8e9-4209-816a-2a967e269a35) while still holding the middle button
8. Connect USB-C cable
9. `cd Linux_for_Tegra`
10. `sudo ./flash.sh "jetson-agx-orin-devkit" "internal"`

**Reply by GJSea** [reply]
Hi! 

Wow, I went through your steps and bingo it all sprung to life with a totally fresh Jetson install :-)

Thanks so much for the rescue!!

**Reply by daniel-amag** [reply]
No worries mate, glad it fired up for you.

**Reply by Eldho1416** [reply]
A small correction here 
  4. `cd ~/nvidia/nvidia_sdk/JetPack_5.1.1_Linux_JETSON_AGX_ORIN_TARGETS/Linux_for_Tegra`
  5. `sudo ./apply_binaries.sh`

Also many thanks for saving my time and anxiety xD

**Comment by Eldho1416** [comment]
Hello,

Are there any instructions to flash jetpack 6?

**Reply by daniel-amag** [reply]
Looks like it should be pretty soon according to the [forum](https://forum.seeedstudio.com/t/j401x-jetpack-6-0-compatibility/273920/36)

**Reply by Eldho1416** [reply]
okay thanks

**Comment by UzairMehmood1** [comment]
Hello Everyone,

I need help with an issue, I want to install Jetpack version 5.1.2. How can I install it on this development kit

Thanks

**Reply by daniel-amag** [reply]
You'd probably want to flash with 5.1.1 (follow my instructions above if you like) and then upgrade to a new minor release by following [Updating to a new point release](https://docs.nvidia.com/jetson/archives/r36.3/DeveloperGuide/SD/SoftwarePackagesAndTheUpdateMechanism.html#updating-to-a-new-minor-release) using r35.4 as your desired release.

**Reply by UzairMehmood1** [reply]
Thank you for your help

**Reply by UzairMehmood1** [reply]
Can I use GMSL cameras on this development kit?

**Comment by UzairMehmood1** [comment]
I have downloaded 
Jetson linux version 35.3.1 to install jetpack version 5.1.1.
Jetpack 5.1.1 flashed successfully, buttwhen I shut it down and power it back on it takes 15 minutes on boot screen, but when I reboot it, it works fine. 
Can anyone guide me about this
Thanks

**Reply by UzairMehmood1** [reply]
I am using this development kit and I have downloaded jetson linux 35.3.1 from this link
https://developer.nvidia.com/embedded/jetson-linux-r3531

**Reply by UzairMehmood1** [reply]
Also, it wont happend when I flash Jetpack 5.0.1

**Comment by GJSea** [comment]
Seems that the JetPack 6.0 peripheral drivers link is broken, any ideas?
https://nv-jetson-images.oss-us-west-1.aliyuncs.com/Orin-AGX-H01/AGX-Orin-H01-JP6.0.zip

**Comment by nashworth1** [comment]
They don't mention it anywhere in the guide, but to flash JetPack 6, your host system needs to have at least Ubuntu 20.04 or you will run into trouble. It will not work using 18.04.

**Reply by daniel-amag** [reply]
In addition, I like using the Docker image of the SDK Manager so you can flash whatever version of JetPack you want without going through the pain of updating or installing different Ubuntu versions to a machine. I use it on a Windows laptop with WSL and can pull or delete the SDK manager with different base OSes as needed. Running the container provides a pretty decent GUI as well. 

For the Seeed Drivers though you'd need to use docker cp or volumes to get the driver files into the container and then exec into it to run the `sudo ./apply_binaries.sh`.  I think this would be done after the downloading and installing of the libraries is complete but before flashing - so when the "Would you like to Flash? Y/N" comes up basically.

**Comment by StrydomMartin** [comment]
Will Jetson 6.1 come out anytime soon? I would like to use Nvidia Metropolis APIs.

**Reply by nfs0619** [reply]
Thanks for reaching out! For information regarding the release of Jetson 6.1, we recommend contacting our technical support team at techsupport@seeed.io. They can provide the most up-to-date details on this.

**Comment by imed-r** [comment]
I'd like to apply the Seeed Studio's peripheral drivers on top of a yocto-built Tegraflash instead of the downloads from step 1. Won't overwriting the `Linux_for_Tegra/kernel/Image` file undo my kernel customizations in Yocto? What should I do instead?

**Reply by nfs0619** [reply]
Hello,

You can find the answer to your question in the following GitHub issue: [Linux_for_Tegra/issues/28](https://github.com/Seeed-Studio/Linux_for_Tegra/issues/28). We have provided the answer there.

If you have any other questions, feel free to contact our technical support at [techsupport@seeed.io](mailto:techsupport@seeed.io).

**Comment by NU-iz-da-fam** [comment]
Hi team, I would like to ask how to apply Jetpack 5.1.1 now. Because our project requires ubuntu 20. 
I followed exactly the instruction but always facing USB timeout error when applying 5.1.1 and your peripherals drivers. 
But, I could install Jetpack 6.x. 
I realized that you had some updates on Orin on November 25, 2024, did they affect flashing procedure?

Thanks in advance!

**Reply by nfs0619** [reply]
Hi, please first confirm whether you're using a **native Ubuntu environment** for flashing. We do not recommend flashing from a virtual machine, as it can cause USB timeout errors.

If the issue persists, we recommend reaching out to our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)** for further assistance.

**Reply by NU-iz-da-fam** [reply]
Thank you for your quick reply!
I use native Ubuntu environment as host PC.
with Jetpack 5.1.1 is Ubuntu 20.04
with Jetpack 6.x, I use Ubuntu 22.04 (same cable, and in recovery mode)
This is also what Nvidia recommends in their documentations.

It's weird that I can flash latest Jetpack. I also saw same issue in Seed Studio Forum.
Do you have any comments?

**Reply by nfs0619** [reply]
This might require specific driver patches. We recommend contacting **[techsupport@seeed.io](mailto:techsupport@seeed.io)** for more detailed support—they can help you troubleshoot further.

**Reply by NU-iz-da-fam** [reply]
Hi, I solved the problem with technical supports from Seeed Studio, and with their permission, I would like to share the solution. 

There is PCN changes on Jetson AGX Orin made by NVidia, but it's not Seeed's PCN changes. 

With Jetpack 5.1.1, beside materials mentioned in Wiki, we need to download [Overlay_PCN210361_PCN210100_r35.3.1.tbz2] at bottom of the page Jetson Linux 35.3.1.
After extracting Jetson_Linux, we need to extract Overlay, copy all files and folders to Linux_For_Tegra (Merge, not Skip).

Then, do remaining steps in Wiki, and my Orin has been flashed succesfully. 

Notes: my Orin is brand new, not sure if this phenomena occured with your current device.
Hope it help!

**Comment by nicholas-presien** [comment]
Hey Guys,

When will the Jetpack 6.2 peripherals drivers be released for the H01 devkit? Can we use the Jetpack 6.1 peripherals drivers for Jetpack 6.2?

**Reply by nfs0619** [reply]
Hi, we are currently arranging updates regarding this, and the documentation will be updated soon. If you need it urgently, please feel free to contact our technical support at [techsupport@seeed.io](mailto:techsupport@seeed.io).

**Reply by MaCo117** [reply]
Hello @nfs0619, any news on the Jetpack 6.2? Thanks!

**Reply by nfs0619** [reply]
Thanks for following up! Here is the relevant peripheral driver package for JetPack 6.2 on the H01 devkit. The flashing process for JP 6.2 is similar to that of JP 6.1, so you can refer to the JP 6.1 flashing instructions in our documentation as a guide.

[605_jp62.tar.gz](https://github.com/user-attachments/files/21427863/605_jp62.tar.gz)

**Comment by rutgermakediff** [comment]
We had a lot of problems working with GPIO until we realized the pin-out markings on the casing are incorrect. Pin 1 was on the right side, 40 on the left side. If you have any inexplicable problems with GPIO, double check the markings on the board itself by opening up the unit and verifying the location of pin 1.

Also mentioned in https://forums.developer.nvidia.com/t/gpio-pins-on-jetson-agx-orin-not-working/296462, but we only found that post after realizing what was wrong :)

**Reply by nfs0619** [reply]
Thank you for your feedback. We will investigate this issue promptly.

**Comment by ghost** [comment]
We had the same issue on both Orin AGX and Xavier AGX kits from Seeed. Just took a burnt CAN transceiver to figure out what was going on. We were surprised that the same mistake was present in both kits.

-g

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by NODeeJay** [comment]
Is there any plan to release 36.4.4? 
And maybe this could move to the Tegra_for_linux repo?

**Reply by ReKST-117** [reply]
Please contact us at techsupport@seeed.io for more details

**Comment by KoreTeknology** [comment]
**Do you know what retaining screw type is used to attach the SSD M.2 drive to this kit?** 
As i bough the samsung Drive without screw, i am wondering what size and type i need. It looks like the Seeedstudio board has a specific screw size on this feature.. thx

**Reply by bleaaach** [reply]
Hi, an M2 screw will work.

**Reply by NU-iz-da-fam** [reply]
Your Orin should be delivered with the suitable screw, I think.

**Reply by NODeeJay** [reply]
Sadly, it was not in my case. The NVIDIA Devkit yes, but not the seeedstudio H01

**Comment by nashworth1** [comment]
Is there a pre-built package for the Realtek BL-M8822CP1 wifi module for Jetpack 6.0 R36.3?

**Reply by bleaaach** [reply]
Hello, it comes with this specific driver package.

**Comment by fahriwps** [comment]
Does this only install the Jetson OS?
Do I need to use SDK Manager to install the Jetson SDK components and the CUDA Toolkit?

**Reply by bleaaach** [reply]
Hi, 
you can run this command to install these components:
**sudo apt install nvidia-jetpack**
<img width="765" height="426" alt="企业微信截图_17634559284337" src="https://github.com/user-attachments/assets/d93cd85c-4233-48f1-81e9-eb9a1fe69afb" />

**Comment by chelseaohara** [comment]
Sharing this in case anyone else runs into this issue. We flashed our seeedstudio jetson (2024) with Jetpack 6.2, 6.1, and 6.0 in an attempt to upgrade to Ubuntu 22, encountering the same problem each time with a Configure USB script failing after the initial restart. Jammy configuration would launch, but without the proper USB overlay, rendering the mouse and keyboard useless and setup impossible. 

By chance, I discovered that powering off/on and booting from the BOOT LOADER menu makes the Configure USB script run and the overlays map correctly. No issue with setup otherwise, but will update if there are any other issues with the upgrade encountered. Cheers

**Comment by NODeeJay** [comment]
Can we have JetPack 7 please? vLLM does not run well and ollama could also see an update for a long time...

https://forums.developer.nvidia.com/t/jetpack-7-2-jetson-linux-r39-2-on-jetson-agx-orin-developer-kit-getting-started-and-feedback-thread/372156?utm_source=chatgpt.com

**Reply by bleaaach** [reply]
Hi, it's scheduled to be released next month.

**Reply by NODeeJay** [reply]
Hi Lorraine,

that's really good news!

Will there be also a USB stick update function like for the NVIDIA devkits or do we need to access the device directly with another Ubuntu instance?

**Reply by bleaaach** [reply]
Hi, the USB burning function does not support QSPI burning. Version 7.2 still requires an Ubuntu host for programming.
